modules/ingest-geoip/src/main/java/org/elasticsearch/ingest/geoip/DatabaseReaderLazyLoader.java

```diff
@@ -194,6 +198,15 @@ int current() {
     }
 
     DatabaseReader get() throws IOException {
+        if (lastUpdate != 0) {
+            Path fileName = databasePath.getFileName();
+            if (System.currentTimeMillis() - lastUpdate > Duration.ofDays(30).toMillis()) {
+                throw new IllegalStateException("database [" + fileName + "] was not updated for 30 days and is disabled");
+            } else if (System.currentTimeMillis() - lastUpdate > Duration.ofDays(25).toMillis() && warningEmitted.getAndSet(true)) {
+                LOGGER.warn("database [{}] was not updated for over 25 days, ingestion will fail if there is no update for 30 days",
```

Comment:
This can lead to many logs and flooding the disk. Perhaps we should emit just a warning?
(using `HeaderWarning.addWarning(...)`)

Reply:
`&& warningEmitted.getAndSet(true)` was there to prevent flooding but header warning makes sense - it makes code cleaner and it gives user more changes to spot potential problem